tfc-dump.pl

```diff
@@ -115,29 +115,39 @@
 my @vs_names;
 my @vs_ids;
 my $pg_size = 100;
+my $pg_num = 1;
 my $total_count;
 my $tmpfile = `mktemp`;
 chomp($tmpfile);
 
-$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/varsets?page%5Bsize%5D=${pg_size}\"";
-$curl_cmd   = "curl $curl_headers --output $tmpfile $curl_query";
-system($curl_cmd);
-
-# Get the Variable Set names
-
-$jq_cmd   = "cat $tmpfile | jq '.data[].attributes.name'";
-@vs_names = `$jq_cmd`;
-# Remove the double quotes in each element of the array.
-grep($_ =~ s/"//g && 0, @vs_names);	# Programming Perl, p. 221, 1990
-chomp(@vs_names);
-
-# Get the Variable Set IDs
+do {
+	$curl_query = "\"https://app.terraform.io/api/v2/organizations/${tfc_org_name}/varsets?page%5Bsize%5D=${pg_size}&page%5number%5D=${pg_num}\"";
+	$curl_cmd   = "curl $curl_headers --output $tmpfile $curl_query";
+	system($curl_cmd);
 
-$jq_cmd = "cat $tmpfile | jq '.data[].id'";
-@vs_ids = `$jq_cmd`;
-# Remove the double quotes in each element of the array.
-grep($_ =~ s/"//g && 0, @vs_ids);	# Programming Perl, p. 221, 1990
-chomp(@vs_ids);
+	# Get the Variable Set names
+	$jq_cmd   = "cat $tmpfile | jq '.data[].attributes.name'";
+	my @page_names = `$jq_cmd`;
+
+	# Remove the double quotes in each element of the array.
+	grep($_ =~ s/"//g && 0, @page_names);	# Programming Perl, p. 221, 1990
+	chomp(@page_names);
+	push(@vs_names, @page_names);
+
+	# Get the Variable Set IDs from this page
+    $jq_cmd = "cat $tmpfile | jq '.data[].id'";
+    my @page_ids = `$jq_cmd`;
+    grep($_ =~ s/"//g && 0, @page_ids);
```

Comment:
Oh yuck. Missed this one. Same problem as above.

Reply:
These dark-magic greps are my first set of required changes.

Reply:
Given you didn't do this yet.
```suggestion
    @page_ids = map { s/"//gr } @page_ids;
```